docs/getting-started/fundamentals-block-development/javascript-in-the-block-editor.md

```diff
@@ -0,0 +1,47 @@
+
+Browsers cannot interpret or run ESNext and JSX syntaxes, so a transformation step is needed to convert these syntaxes to code that browsers can understand.
+
+["webpack"](https://webpack.js.org/concepts/why-webpack/) is a pluggable tool that processes JavaScript and creates a compiled bundle that runs in a browser. ["babel"](https://babeljs.io/) transforms JavaScript from one format to another. Babel is used as a plugin to webpack to transform both ESNext and JSX to production-ready JavaScript.
+
+[`@wordpress/scripts`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/) package abstracts these libraries away to standardize and simplify development, so you won’t need to handle the details for configuring webpack or babel. Check the [Get started with wp-scripts](https://developer.wordpress.org/block-editor/getting-started/devenv/get-started-with-wp-scripts/) intro guide.
+
+Among other things, with `wp-scripts` package you can use Javascript modules to distribute your code among different files and get a few bundled files at the end of the build process (see [example](https://github.com/WordPress/block-development-examples/tree/trunk/plugins/data-basics-59c8f8)).
+
+With the [proper `package.json` scripts](https://developer.wordpress.org/block-editor/getting-started/devenv/get-started-with-wp-scripts/#basic-usage) you can launch the build process with `wp-scripts`  in production and development mode:
+
+- **`npm run build` for "production" mode build** - This process [minifies the code](https://developer.mozilla.org/en-US/docs/Glossary/Minification) down so it downloads faster in the browser. 
+- **`npm run start` for "development" mode build**  - This process does not minify the code of the bundled files, provides [source maps files](https://firefox-source-docs.mozilla.org/devtools-user/debugger/how_to/use_a_source_map/index.html) for them, and additionally continues a running process to watch the source file for more changes and rebuilds as you develop.
+
+## Javascript without a build process
+
+For code developments with few requirements (especially those not requiring JSX) using Javascript without a build process may be another good option. 
+
+Without a build process you access the methods directly from the `wp` global object and you need to enqueue manually the script.
+
+`wp` is the main WordPress JavaScript object available a [global variable](https://developer.mozilla.org/en-US/docs/Glossary/Global_variable). It is provisionned with different properties depending on what page is loaded. In the Block Editor context 
```

Comment:
```suggestion
Any of the WordPress bundled packages can be accessed through the `wp` [global variable](https://developer.mozilla.org/en-US/docs/Glossary/Global_variable). Every script that wants to use any of these bundled packages is responsible for adding the handle or that package to the dependency array when registered. 

So for example if a script wants to use the `RichText` component out of the block editor package `wp-block-editor` would need to get added to the dependency array to ensure that `wp.blockEditor.RichText` is defined then the script tries to access it.
```

Reply:
Whilst it is true that depending on which editor you land in the packages available differs, that is not actually something you as a developer should be too concerned with. 

There is no guarantee that something will be available through the global wp object unless your script specifically ensures that the package has been added to the dependency array. 

Under the hood each of these packages are just other JS files that get loaded on the page. And adding them to the dependency array ensures that they load before your script so that you can safely use them. 

If you don't add them to your dependency array a further WordPress version may drop the dependency or a certain package somewhere and then your code breaks unexpectedly.

Reply:
This by the way is also true if you use a build process. Just that there is a webpack plugin that automatically generates a PHP file that contains the dependency array so you don't manually have to maintain it. It just happens automatically for you